File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-2_Gestion-Employes/Implementation-Java/Entreprise.java
import java.util.ArrayList;
import java.util.List;

public class Entreprise {
    private List<Employe> employes;

    public Entreprise() {
        this.employes = new ArrayList<>();
    }

    public void ajouterEmploye(Employe employe) {
        employes.add(employe);
    }

    public double getSalaireTotal() {
        double salaireTotal = 0;
        for (Employe employe : employes) {
            salaireTotal += employe.calculerSalaire();
        }
        return salaireTotal;
    }

    public String organigramme() {
        String organigram = "Organigramme de l'entreprise:\n";
        for (Employe employe : employes) {
            organigram += employe.afficherSuperieur() + "\n";
            organigram += employe.afficherSubordonnes() + "\n";
        }
        return organigram;
    }
}

File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-2_Gestion-Employes/Implementation-Cpp/Entreprise.cpp
#include "Entreprise.h"
#include "Employe.h"

Entreprise::Entreprise() {}

void Entreprise::ajouterEmploye(Employe * employe) {
    employes.push_back(employe);
}

double Entreprise::getSalaireTotal() const {
    double salaireTotal = 0;
    for (const Employe * employe : employes) {
        salaireTotal += employe->calculerSalaire();
    }
    return salaireTotal;
}

string Entreprise::organigramme() const {
    string organigram = "Organigramme de l'entreprise:\n";
    for (const Employe * employe : employes) {
        organigram += employe->afficherSuperieur() + "\n";
        organigram += employe->afficherSubordonnes() + "\n";
    }
    return organigram;
}